conbench/entities/benchmark_result.py

```diff
@@ -435,10 +435,15 @@ def _single_value_summary(self) -> float:
         if self.is_failed():
             return math.nan
 
-        # Should be implied by the outcome of self.is_failed(), encode this.
-        assert self.mean is not None
+        if self.mean is None:
+            # See https://github.com/conbench/conbench/issues/1169 -- Legacy
+            # database might have mean being None _despite the benchmark not
+            # being failed_. Because of a temporary logic error. Let's remove
+            # this code path again for sanity. Since is_failed() returned False
+            # we know that `self.data` has only numbers, and at least one.
+            assert self.data is not None
+            return float(statistics.mean(self.data))
```

Comment:
Not beautiful, but with all the commentary and the is_failed() abstraction I think this is easy enough to reason about in the future. 

To me, the biggest takeaway is that we want to enhance our test coverage, because so far we had so many non-exercised scenarios in the test suite: https://github.com/conbench/conbench/issues/1184